src/main/java/net/tropicraft/core/common/dimension/feature/RainforestVinesFeature.java

```diff
@@ -38,7 +38,7 @@ public boolean generate(ISeedReader world, ChunkGenerator generator, Random rand
                         mutablePos.move(direction);
                         BlockState attaching = world.getBlockState(mutablePos);
                         if ((attaching.getBlock() == Blocks.GRASS_BLOCK && rand.nextInt(4) == 0) || attaching.isIn(BlockTags.LEAVES)) {
-                            if (direction != Direction.DOWN && VineBlock.hasEnoughSolidSide(world, mutablePos, direction)) {
+                            if (direction != Direction.DOWN && (attaching.isIn(BlockTags.LEAVES) || VineBlock.hasEnoughSolidSide(world, mutablePos, direction))) {
```

Comment:
I believe this should rather use `VineBlock.canAttachTo` which accounts for leaves